bin/network-monitor/src/faucet.rs

```diff
@@ -0,0 +1,240 @@
+                warn!("Faucet test failed: {}", e);
+            },
+        }
+
+        let test_duration_ms = start_time.elapsed().as_millis() as u64;
+
+        let test_details = FaucetTestDetails {
+            test_duration_ms,
+            success_count,
+            failure_count,
+            last_tx_id: last_tx_id.clone(),
+            last_note_id: last_note_id.clone(),
+            challenge_difficulty: last_challenge_difficulty,
+        };
+
+        let status = ServiceStatus {
+            name: "Faucet".to_string(),
+            status: if success_count > 0 || failure_count == 0 {
+                Status::Healthy
+            } else {
+                Status::Unhealthy
+            },
+            last_checked: current_time,
+            error: None,
+            details: ServiceDetails::FaucetTest(test_details),
+        };
+
+        // Send the status update; exit if no receivers (shutdown signal)
+        if status_sender.send(status).is_err() {
+            info!("No receivers for faucet status updates, shutting down");
+            return Ok(());
+        }
+    }
+}
+
+/// Performs a complete faucet test by requesting a `PoW` challenge and submitting the solution.
+///
+/// # Arguments
+///
+/// * `client` - The HTTP client to use.
+/// * `faucet_url` - The URL of the faucet service.
+///
+/// # Returns
+///
+/// The response from the faucet if successful, or an error if the test fails.
+async fn perform_faucet_test(
+    client: &Client,
+    faucet_url: &Url,
+) -> anyhow::Result<(GetTokensResponse, u64)> {
+    // Use a test account ID - convert to AccountId and format properly
+    let account_id = AccountId::try_from(ACCOUNT_ID_SENDER)
+        .context("Failed to create AccountId from test constant")?;
+
+    let account_id = account_id.to_string();
+    debug!("Generated account ID: {} (length: {})", account_id, account_id.len());
+
+    // Step 1: Request PoW challenge
+    let pow_url = faucet_url.join("/pow")?;
+    let response = client.get(pow_url).query(&[("account_id", &account_id)]).send().await?;
+
+    let response_text = response.text().await?;
+    debug!("Faucet PoW response: {}", response_text);
+
+    let challenge_response: PowChallengeResponse = serde_json::from_str(&response_text)
+        .with_context(|| format!("Failed to parse PoW response: {response_text}"))?;
+
+    debug!(
+        "Received PoW challenge: target={}, challenge={}...",
+        challenge_response.target,
+        &challenge_response.challenge[..16.min(challenge_response.challenge.len())]
+    );
+
+    // Step 2: Solve the PoW challenge
+    let nonce = solve_pow_challenge(&challenge_response.challenge, challenge_response.target)
+        .context("Failed to solve PoW challenge")?;
+
+    debug!("Solved PoW challenge with nonce: {}", nonce);
+
+    // Step 3: Request tokens with the solution
+    let tokens_url = faucet_url.join("/get_tokens")?;
+    let asset_amount = 1_000_000u64; // 1 token with 6 decimals
```

Comment:
Could we create a constant for this? (this way it would be at the top of the file)